graphdatascience/query_runner/progress/static_progress_provider.py

```diff
@@ -0,0 +1,27 @@
+from typing import Optional
+
+from .progress_provider import ProgressProvider, TaskWithProgress
+
+
+class StaticProgressStore:
```

Comment:
this class isn't meant to stick around, but you never know... a unit test or two could be useful I think

Reply:
although unit tests of these static singletons can be a bit annoying